<launch>
  <!-- Shared Arguments -->
  <arg name="map_yaml" default="true" doc="filepath to previously captured map (required)" />
  <arg name="debug_directory" default="$(env HELLO_FLEET_PATH)/debug/" doc="path to directory where debug planning images/logs are stored" />
  <arg name="dryrun" default="false" doc="whether run perception/planning without actually moving" />
  <arg name="rviz" default="true" doc="whether to show Rviz" />
  <arg name="respeaker" value="false" />

  <!-- STRETCH DRIVER -->
  <param name="/stretch_driver/broadcast_odom_tf" type="bool" value="true" />
  <param name="/stretch_driver/mode" type="string" value="navigation" />
  <include file="$(find stretch_core)/launch/stretch_driver.launch" />

  <!-- CENTERED BASE LINK -->
  <node name="centered_base_link_tf_publisher" pkg="tf" type="static_transform_publisher" args="-0.1 0 0 0 0 0 1 /base_link /centered_base_link 100" />

  <!-- LASER RANGE FINDER -->
  <include file="$(find stretch_core)/launch/rplidar.launch" />

  <!-- MAP SERVER -->
  <node name="map_server" pkg="map_server" type="map_server" args="$(env HELLO_FLEET_PATH)/maps/riot_lab.yaml" />

  <!-- LOCALIZATION -->
  <include file="$(find amcl)/examples/amcl_diff.launch" />

  <!-- MAPPING -->
  <node name="funmap" pkg="stretch_funmap" type="funmap" output="screen">
    <param name="debug_directory" type="string" value="$(arg debug_directory)" />
  </node>

  <!-- WAYPOINTS NAVIGATION -->
  <node name="waypoints_navigation" pkg="stretch_demos" type="waypoints_navigation.py" output="screen" />

  <!-- VISUALIZE -->
  <!-- Checks if RViz should be launched and decides which config to use based on the 'rviz' argument -->
  <group if="$(arg rviz)">
    <node name="rviz_navigation" pkg="rviz" type="rviz" output="log" args="-d $(find stretch_navigation)/rviz/navigation.rviz" />
  </group>
</launch>
